cmake/Modules/FindDPDK.cmake

```diff
@@ -0,0 +1,252 @@
+else()
+  find_path(
+    dpdk_config_INCLUDE_DIR rte_config.h
+    PATH_SUFFIXES dpdk include)
+  find_path(
+    dpdk_common_INCLUDE_DIR rte_common.h
+    PATH_SUFFIXES dpdk include)
+  set(dpdk_INCLUDE_DIRS "${dpdk_config_INCLUDE_DIR}")
+  if(NOT dpdk_config_INCLUDE_DIR STREQUAL dpdk_common_INCLUDE_DIR)
+    list(APPEND dpdk_INCLUDE_DIRS "${dpdk_common_INCLUDE_DIR}")
+  endif()
+endif()
+
+# Order is important ! Taken from the DPKT pkg-config file
+set(components
+    common_cpt
+    common_dpaax
+    common_iavf
+    common_octeontx
+    common_octeontx2
+    common_sfc_efx
+    bus_dpaa
+    bus_fslmc
+    bus_ifpga
+    bus_pci
+    bus_vdev
+    bus_vmbus
+    common_qat
+    mempool_bucket
+    mempool_dpaa
+    mempool_dpaa2
+    mempool_octeontx
+    mempool_octeontx2
+    mempool_ring
+    mempool_stack
+    net_af_packet
+    net_ark
+    net_atlantic
+    net_avp
+    net_axgbe
+    net_bond
+    net_bnxt
+    net_cxgbe
+    net_dpaa
+    net_dpaa2
+    net_e1000
+    net_ena
+    net_enetc
+    net_enic
+    net_failsafe
+    net_fm10k
+    net_i40e
+    net_hinic
+    net_hns3
+    net_iavf
+    net_ice
+    net_igc
+    net_ixgbe
+    net_kni
+    net_liquidio
+    net_memif
+    net_netvsc
+    net_nfp
+    net_null
+    net_octeontx
+    net_octeontx2
+    net_pcap
+    net_pfe
+    net_qede
+    net_ring
+    net_sfc
+    net_softnic
+    net_tap
+    net_thunderx
+    net_txgbe
+    net_vdev_netvsc
+    net_vhost
+    net_virtio
+    net_vmxnet3
+    raw_dpaa2_cmdif
+    raw_dpaa2_qdma
+    raw_ioat
+    raw_ntb
+    raw_octeontx2_dma
+    raw_octeontx2_ep
+    raw_skeleton
+    crypto_bcmfs
+    crypto_caam_jr
+    crypto_dpaa_sec
+    crypto_dpaa2_sec
+    crypto_nitrox
+    crypto_null
+    crypto_octeontx
+    crypto_octeontx2
+    crypto_scheduler
+    crypto_virtio
+    compress_octeontx
+    regex_octeontx2
+    vdpa_ifc
+    event_dlb
+    event_dlb2
+    event_dpaa
+    event_dpaa2
+    event_octeontx2
+    event_opdl
+    event_skeleton
+    event_sw
+    event_dsw
+    event_octeontx
+    baseband_null
+    baseband_turbo_sw
+    baseband_fpga_lte_fec
+    baseband_fpga_5gnr_fec
+    baseband_acc100
+    node
+    graph
+    bpf
+    flow_classify
+    pipeline
+    table
+    port
+    fib
+    ipsec
+    vhost
+    stack
+    security
+    sched
+    reorder
+    rib
+    regexdev
+    rawdev
+    pdump
+    power
+    member
+    lpm
+    latencystats
+    kni
+    jobstats
+    ip_frag
+    gso
+    gro
+    eventdev
+    efd
+    distributor
+    cryptodev
+    compressdev
+    cfgfile
+    bitratestats
+    bbdev
+    acl
+    timer
+    hash
+    metrics
+    cmdline
+    pci
+    ethdev
+    meter
+    net
+    mbuf
+    mempool
+    rcu
+    ring
+    eal
+    telemetry
+    kvargs)
+
+# for collecting dpdk library targets, it will be used when defining dpdk::dpdk
+set(_dpdk_libs)
+# for list of dpdk library archive paths
+set(dpdk_LIBRARIES "")
+foreach(c ${components})
```

Comment:
I think it also makes the build very slow... I ran it locally and it took the examples a long time to build (around 30 mins, maybe more 😨 )

Reply:
I thought it's only on my machine, but then I compared the current build on `dev` to the new build with CMake:

on `dev` the build with DPDK 20.11 takes ~5-5:30 mins: https://github.com/seladb/PcapPlusPlus/runs/8196725078?check_suite_focus=true

But in `cmake-light` the build with 20.11 takes ~8:30 mins: https://github.com/seladb/PcapPlusPlus/runs/8196347792?check_suite_focus=true

Reply:
should be better now.